src/Bicep.Core/TypeSystem/UnassignableType.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bicep.Core.Diagnostics;
+
+namespace Bicep.Core.TypeSystem
+{
+    public class UnassignableTypeSymbol : TypeSymbol
+    {
+        private const string ErrorTypeName = "error";
+        private readonly ImmutableArray<ErrorDiagnostic> errors;
+        private readonly TypeKind kind;
+
+        public UnassignableTypeSymbol(string name, TypeKind kind)
+            : base(name)
+        {
+            this.kind = kind;
+
+            // if errors are not initialized, initialize it as an empty array
```

Comment:
I don't think you need this check - you're in the constructor, so errors will always be uninitialized.